source/fuzz/transformation_duplicate_region_with_selection.cpp

```diff
@@ -0,0 +1,566 @@
+      // The continue target is a single-entry, single-exit region. Therefore,
+      // if the continue target is the exit block, the region might not contain
+      // the loop header.
+      if (continue_target != exit_block &&
+          region_set.count(&block) != region_set.count(continue_target)) {
+        return false;
+      }
+    }
+  }
+
+  // Get the maps from the protobuf.
+  std::map<uint32_t, uint32_t> original_label_to_duplicate_label =
+      fuzzerutil::RepeatedUInt32PairToMap(
+          message_.original_label_to_duplicate_label());
+
+  std::map<uint32_t, uint32_t> original_id_to_duplicate_id =
+      fuzzerutil::RepeatedUInt32PairToMap(
+          message_.original_id_to_duplicate_id());
+
+  std::map<uint32_t, uint32_t> original_id_to_phi_id =
+      fuzzerutil::RepeatedUInt32PairToMap(message_.original_id_to_phi_id());
+
+  for (auto block : region_set) {
+    auto label =
+        ir_context->get_def_use_mgr()->GetDef(block->id())->result_id();
+    // The label of every block in the region must be present in the map
+    // |original_label_to_duplicate_label|.
+    if (original_label_to_duplicate_label.count(label) == 0) {
+      return false;
+    }
+    auto duplicate_label = original_label_to_duplicate_label[label];
+    // Each id assigned to labels in the region must be distinct and fresh.
+    if (!duplicate_label ||
+        !CheckIdIsFreshAndNotUsedByThisTransformation(
+            duplicate_label, ir_context, &ids_used_by_this_transformation)) {
+      return false;
+    }
+    for (auto instr : *block) {
+      if (!instr.HasResultId()) {
+        continue;
+      }
+      // Every instruction with a result id in the region must be present in the
+      // map |original_id_to_duplicate_id|.
+      if (original_id_to_duplicate_id.count(instr.result_id()) == 0) {
+        return false;
+      }
+      auto duplicate_id = original_id_to_duplicate_id[instr.result_id()];
+      // Id assigned to this result id in the region must be distinct and fresh.
+      if (!duplicate_id ||
+          !CheckIdIsFreshAndNotUsedByThisTransformation(
+              duplicate_id, ir_context, &ids_used_by_this_transformation)) {
+        return false;
+      }
+      if (&instr == &*exit_block->tail() ||
+          fuzzerutil::IdIsAvailableBeforeInstruction(
+              ir_context, &*exit_block->tail(), instr.result_id())) {
+        // Every instruction with a result id available at the end of the region
+        // must be present in the map |original_id_to_phi_id|.
+        if (original_id_to_phi_id.count(instr.result_id()) == 0) {
+          return false;
+        }
+        // Using pointers with OpPhi requires capability VariablePointers.
+        if (ir_context->get_type_mgr()->GetType(instr.type_id())->AsPointer() &&
+            !ir_context->get_feature_mgr()->HasCapability(
+                SpvCapabilityVariablePointers)) {
+          return false;
+        }
```

Comment:
This could end up being rather limiting.  Our decision to create OpPhis for all available ids, whether or not they are actually used, is good in general, but may stop this transformation from applying if there are pointer ids with this property, and if those pointer ids are not actually used after the region.

For pointers, and where VariablePointers is not available, it would be better if we could allow the transformation to proceed with *no* OpPhi if it's the case that the pointer is not actually used after the region.

Can you open an issue for this and add a TODO(...) comment here?  We can address this in a separate PR.